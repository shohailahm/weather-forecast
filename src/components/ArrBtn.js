import React from 'react'
import Grid from '@material-ui/core/Grid';
import { NavigateBefore, NavigateNext } from '@material-ui/icons';
export default class ArrBtn extends React.Component{
   render(){
       return(
        <Grid container>
        <Grid item xs={4}>
          {this.props.showLeft?<div onClick={this.props.prev}><NavigateBefore /></div>:null}
        </Grid>
        <Grid item xs={4}>
          
        </Grid>
        <Grid item xs={4}>
          {this.props.showRight?<div onClick={this.props.next}><NavigateNext onClick={()=>this.props.next()}/></div>:null} 
          </Grid>
        </Grid>
       )
   }

}