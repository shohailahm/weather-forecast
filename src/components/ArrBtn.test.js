import React from 'react';
import { shallow } from 'enzyme';
import ArrBtn from './ArrBtn.js';

describe('ArrowButton', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<ArrBtn />);
      expect(wrapper).toMatchSnapshot();
      // On the first run of this test, Jest will generate a snapshot file automatically.
    });
  });

  it('disables the button element', () => {
    const wrapper = shallow(<ArrBtn />);
    const buttonElement = wrapper.find('.Left');
    expect(buttonElement.props('showLeft').item).toBe(true);
  
  });
  
//   it('does not render the button element when [certain state is true]', () => {
//    });

// describe('BaseButton', () => {
//     describe('when user clicks button', () => {
//       it ('calls correct function to save the information', () => {
//         const onButtonClickMock = jest.fn();
//         const wrapper = shallow(
//           <ArrBtn
//             onButtonClick={onButtonClickMock}
//           />,
//         );
//         const buttonElement = wrapper.find('.Right'); // step 1 above
//         buttonElement.simulate('click'); // step 2
        
//         expect(onButtonClickMock).toHaveBeenCalledTimes(1); // step 3
//         expect(onButtonClickMock).toHaveBeenCalledWith(true);
//       });
//     });
//   });