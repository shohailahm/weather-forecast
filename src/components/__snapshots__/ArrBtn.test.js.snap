// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrowButton renders correctly 1`] = `ShallowWrapper {}`;
